lib/mayaUsd/render/vp2RenderDelegate/mesh.cpp

```diff
@@ -365,6 +365,20 @@ void HdVP2Mesh::Sync(
         return;
     }
 
+    // We don't update the repr if it is hidden by the render tags (purpose)
+    // of the ProxyRenderDelegate. In additional, we need to hide any already
+    // existing render items because they should not be drawn.
+    auto* const          param = static_cast<HdVP2RenderParam*>(_delegate->GetRenderParam());
+    ProxyRenderDelegate& drawScene = param->GetDrawScene();
+    if (!drawScene.DrawRenderTag(delegate->GetRenderIndex().GetRenderTag(GetId()))) {
+        _HideAllDrawItems(reprToken);
+        // clearing visibility here is wrong. It is a work around to handle purpose changing
+        // marking visibility dirty instead of render tag. See UsdImagingGprimAdapter::ProcessPropertyChange
+        // for where this happens.
+        *dirtyBits &= ~( HdChangeTracker::DirtyRenderTag | HdChangeTracker::DirtyVisibility );
+        return;
```

Comment:
This is the risky bit of the change. We don't have much experience or tests leaving items dirty & hidden. I tested a scene with animation where I change the purpose of the items so they are hidden, change time, and change the purpose so they are drawn, and that simple workflow is correct.

Reply:
This pattern will have to be repeated in the other HdVP2 shapes because they all have the same problem. Ideally we can find a way for the different shapes to share code.